src/EFCore.MySql/ValueGeneration/Internal/MySqlSequentialGuidValueGenerator.cs

```diff
@@ -42,21 +65,24 @@ public override Guid Next(EntityEntry entry)
                     Array.Reverse(tickBytes);
                 }
 
-                Buffer.BlockCopy(tickBytes, 0, guidBytes, 0, 8);
-                Buffer.BlockCopy(randomBytes, 0, guidBytes, 8, 8);
+                Buffer.BlockCopy(tickBytes, 0, guidBytes, 0, 6);
+                guidBytes[6] = (byte)(ticksAndVersion << 8 >> 8);
+                guidBytes[7] = (byte)(ticksAndVersion >> 8);
+                guidBytes[8] = ticksAndVariant;
+                Buffer.BlockCopy(randomBytes, 0, guidBytes, 9, 7);
```

Comment:
Please change the rest of the `Buffer.BlockCopy` calls to simple assignments as well. There is probably not much advantage using them over assignments here anymore.